packages/amplify-graphql-transformer-core/src/utils/function-runtime.ts

```diff
@@ -27,40 +33,44 @@ type RuntimeSpecificFunctionProps = {
   runtime?: CfnFunctionConfiguration.AppSyncRuntimeProperty;
   /** The function code as a string. Used for JavaScript runtimes. */
   code?: string;
+  /** The S3 location of the function code. Used for JavaScript runtimes with S3-stored templates. */
+  codeS3Location?: string;
 };
 
 /**
  * Retrieves the runtime-specific function properties based on the provided configuration.
  *
  * @param scope - The construct scope.
  * @param props - An object containing the function configuration properties.
- * @param props.requestMappingTemplate - The request mapping template provider.
- * @param props.responseMappingTemplate - The response mapping template provider.
+ * @param props.mappingTemplate - The function runtime template, either JSRuntimeTemplate or VTLRuntimeTemplate.
  * @param props.runtime - AppSync resolver function runtime configuration.
  * @param props.api - The GraphQL API instance.
  * @returns An object with runtime-specific function properties.
  */
 export const getRuntimeSpecificFunctionProps = (
   scope: Construct,
   props: {
-    requestMappingTemplate: MappingTemplateProvider;
-    responseMappingTemplate: MappingTemplateProvider;
+    mappingTemplate: FunctionRuntimeTemplate;
     runtime?: CfnFunctionConfiguration.AppSyncRuntimeProperty;
     api: GraphQLApi;
   },
 ): RuntimeSpecificFunctionProps => {
-  const { requestMappingTemplate, responseMappingTemplate, runtime, api } = props;
+  const { mappingTemplate, runtime, api } = props;
 
-  const requestTemplateLocation = requestMappingTemplate.bind(scope, api.assetProvider);
-  const responseTemplateLocation = responseMappingTemplate.bind(scope, api.assetProvider);
-
-  if (isJsResolverFnRuntime(runtime)) {
-    return {
-      runtime,
-      code: requestTemplateLocation + '\n\n' + responseTemplateLocation,
-    };
+  if (isJsRuntimeTemplate(mappingTemplate)) {
+    const { codeMappingTemplate } = mappingTemplate;
+    if (!isJsResolverFnRuntime(runtime)) {
+      throw new Error('Runtime provided with codeMappingTemplate is not a JavaScript runtime');
+    }
+    const codeTemplateLocation = codeMappingTemplate.bind(scope, api.assetProvider);
+    return codeMappingTemplate.type === MappingTemplateType.INLINE
```

Comment:
As far as I can tell, the `instanceof InlineTemplate` check conducted for VTL resolvers below isn't a valid conditional for determining if a mapping template is inline or not. I'm not comfortable changing the behavior for VTL resolvers because I assume downstream things are depending on this behavior. But JS resolvers are a new concept here, so this is doing the "correct" check. Hence the discrepancy in conditionals between JS and VTL mapping templates.